ci/authn-azure/get_system_assigned_identity.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -ex
+
+decode_base64_url() {
+  local len=$((${#1} % 4))
```

Comment:
there is a lot of technical, syntax in this function that I am not familiar with and when looking at the decode_base64_url function, I think it would benefit from an explanation as to what you are trying to accomplish and why you are moding how the length of the result will give the a result an extra `=` or not
For my knowledge:
`#` sign. After looking it up, basically means prefix removal and I assume this translates to prefix removal of the access token?

Why does the length of the result, determine if the result has an extra `=` or not?

Am I understanding correctly that you are translating/replacing all `_-` patterns with  `/+`

Reply:
i didn't go into detail of how this method works, and i don't think it is very important. 

Please take a look again at the script. I re-arranged it to be more readable.

Reply:
👍 on the explanation